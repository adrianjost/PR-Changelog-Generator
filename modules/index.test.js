jest.mock("./context");
const core = require("@actions/core");
const nock = require("nock");

const { owner, repo } = require("./__mocks__/context");
const mockAPICall = (requestPath, mockDataPath) => {
	nock("https://api.github.com")
		.get(requestPath)
		.reply(200, require(mockDataPath));
};

const mockPulls = (ids) => {
	ids.forEach((id) => {
		mockAPICall(
			`/repos/schul-cloud/PR-Changelog-Generator/pulls/${id}`,
			`../mock-data/pull_${id}`
		);
	});
};
mockPulls([3, 4, 5, 6, 7, 8, 9]);
mockAPICall(`/repos/${owner}/${repo}/tags`, "../mock-data/tags");
mockAPICall(
	`/repos/schul-cloud/PR-Changelog-Generator/compare/3f699fb44c46f22cd3ce937a58263d2a468a6e92...74fc76826aefa45551b036524b9a1db05fb51f86`,
	"../mock-data/compare"
);

const run = require("./index");

describe("action", () => {
	beforeEach(() => {
		jest.resetModules();
	});

	it("can be executed", async () => {
		jest.spyOn(core, "info").mockImplementation();
		jest.spyOn(core, "debug").mockImplementation();
		const setOutputMock = jest.spyOn(core, "setOutput").mockImplementation();
		await run();
		expect(setOutputMock.mock.calls).toMatchSnapshot();
	});
});
