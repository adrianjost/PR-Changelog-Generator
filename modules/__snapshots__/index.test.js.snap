// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action can be executed 1`] = `
Array [
  Array [
    "release_tag",
    "v0.0.4",
  ],
  Array [
    "pull_requests",
    "[3,4,6,7,8,9]",
  ],
  Array [
    "changelog",
    "## v0.0.4

### Added

- pull request templated for easier and faster PR creation.
- documentation that you also need to setup the sender repository
- documentation of \`token\` input
- new output \`pull_requests\` added, which holds a list of pull request numbers that where included in the release

### Changed

- renamed \`gh-token\` input to \`token\`

### Removed

- MongoDB is no longer required. Therefore the following inputs where removed
  - \`mongo-uri\`
  - \`mongo-db\`
  - \`mongo-collection\`

### Fixed

- action was not starting because of a wrong require name in \`modules\\\\get-pull_request-merges-between-commits.js\`. Changes \`./repo-context\` to \`./context\`
- an import was missing
",
  ],
]
`;
